elide-core/src/main/java/com/yahoo/elide/core/Path.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2017, Oath Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import com.yahoo.elide.core.exceptions.InvalidValueException;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a path in the entity relationship graph
+ */
+public class Path {
+    @Getter
+    private List<PathElement> pathElements;
+    private static final String PERIOD = ".";
+    private static final String UNDERSCORE = "_";
+    /**
+     * The path taken through data model associations to reference a given field.
+     * eg. author.books.publisher.name
+     */
+    @AllArgsConstructor
+    @ToString
+    @EqualsAndHashCode
+    public static class PathElement {
+        @Getter private Class type;
+        @Getter private String typeName;
+        @Getter private Class fieldType;
+        @Getter private String fieldName;
+    }
+
+    public Path(List<PathElement> pathElements) {
+        this.pathElements = pathElements;
+    }
+
+    public Path(PathElement pathElement) {
+        this(Collections.singletonList(pathElement));
+    }
+
+    public Path(Class<?> entityClass, EntityDictionary dictionary, String dotSeparatedPath) {
+        pathElements = new ArrayList<>();
+        String[] fieldNames = dotSeparatedPath.split("\\.");
+
+        Class<?> currentClass = entityClass;
+        String currentClassName = dictionary.getJsonAliasFor(currentClass);
+        for (String fieldName : fieldNames) {
+            if (dictionary.isRelation(currentClass, fieldName)) {
+                Class<?> relationClass = dictionary.getParameterizedType(currentClass, fieldName);
+                pathElements.add(new PathElement(currentClass, currentClassName, relationClass, fieldName));
+                currentClass = relationClass;
+                currentClassName = fieldName;
+            } else if (dictionary.isAttribute(currentClass, fieldName)
+                    || fieldName.equals(dictionary.getIdFieldName(entityClass))) {
+                Class<?> attributeClass = dictionary.getType(currentClass, fieldName);
+                pathElements.add(new PathElement(currentClass, currentClassName, attributeClass, fieldName));
+            } else {
+                throw new InvalidValueException(entityClass.getSimpleName()
```

Comment:
Is `SimpleName` really what we want here?  Should we use type name (which can be different)